datastew/repository/sqllite.py

```diff
@@ -35,6 +37,34 @@ def store_all(self, model_object_instances: List[Union[Terminology, Concept, Map
         self.session.add_all(model_object_instances)
         self.session.commit()
 
+    def import_data_dictionary(self, data_dictionary: DataDictionarySource, terminology_name: str, embedding_model: Optional[EmbeddingModel] = None):
+        terminology = Terminology(terminology_name, terminology_name)
```

Comment:
I would move the storage logic down. If for example there is any issue with the data dictionary and the function fails at that point, we will have an empty terminology stored here without any links to actual concepts.

If the user then tries to redo the process with a fixed data dictionary, their initial preffered terminology name will already be taken and not available.

Reply:
Definitely makes sense, I did not see it. Thanks for pointing out.